File: src/main/java/jp/ac/morijyobi/equipmentmanagementsystem/bean/entity/Account.java
package jp.ac.morijyobi.equipmentmanagementsystem.bean.entity;

import jp.ac.morijyobi.equipmentmanagementsystem.constant.AccountCategory;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Account {
    public int id;
    public String name;
    public String mail;
    public String password;
    public AccountCategory category;
    public Boolean isEnable;

    public Account cryptPassword() {
        final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
        final String newPassword = passwordEncoder.encode(this.password);

        return new Account(
                this.id,
                this.name,
                this.mail,
                newPassword,
                this.category,
                this.isEnable
        );
    }
}

File: src/main/resources/static/js/account/account.js
class Account {
    #mail;
    #name;
    #category;

    constructor(mail, name, category) {
        if (mail === null) throw new Error('mail is null');
        if (name === null) throw new Error('name is null');
        if (category === null) throw new Error('category is null');

        this.#mail = mail;
        this.#name = name;
        this.#category = category;
    }

    static empty() {
        return new Account('none', 'none', 'none');
    }

    mail() {
        return this.#mail;
    }

    name() {
        return this.#name;
    }

    category() {
        return this.#category;
    }

    toJSON() {
        return {
            id: -1,
            mail: this.#mail,
            name: this.#name,
            // NOTE: 仮パスワード
            password: 'morijyobi',
            category: this.#category,
            isEnable: true
        };
    }
}
